R/driver-snowflake.R

```diff
@@ -5,6 +5,60 @@ NULL
 #' @rdname DBI-classes
 setClass("Snowflake", contains = "OdbcConnection")
 
+#' @description
+#' ## `odbcConnectionColumns()`
+#'
+#' If the catalog, or the schema arguments are NULL, attempt to infer
+#' by querying for CURRENT_DATABASE() and CURRENT_SCHEMA().  We do this
+#' to aid with performance, as the SQLColumns method is more performant
+#' when restricted to a particular DB/schema.
+#' @inheritParams DBI::dbListFields
+#' @rdname Snowflake
+#' @usage NULL
+setMethod("odbcConnectionColumns_", c("Snowflake", "character"),
+  function(conn,
+           name,
+           ...,
+           catalog_name = NULL,
+           schema_name = NULL,
+           column_name = NULL,
+           exact = FALSE) {
+    if(is.null(catalog_name) || is.null(schema_name)) {
+      res <- dbGetQuery(conn, "SELECT CURRENT_DATABASE() AS CAT, CURRENT_SCHEMA() AS SCH")
+      if(is.null(catalog_name) && !is.na(res$CAT[1])) {
+        catalog_name <- res$CAT[1]
+      }
+      if(is.null(schema_name) && !is.na(res$SCH[1])) {
+        schema_name <- res$SCH[1]
+      }
+    }
+    callNextMethod(conn = conn, name = name, ..., catalog_name = catalog_name,
+      schema_name = schema_name, column_name = column_name, exact = exact)
+  }
+)
+
+#' If the catalog, or the schema arguments are NULL, attempt to infer
+#' by querying for CURRENT_DATABASE() and CURRENT_SCHEMA().  We do this
+#' to aid with performance, as the SQLTables method is more performant
+#' when restricted to a particular DB/schema.
```

Comment:
In `?Snowflake` currently, this description is repeated twice, and this block is interpreted as the page title. I would remove the above 4 lines

Reply:
Done - thanks!